src/_gettsim/shared.py

```diff
@@ -3,6 +3,9 @@
 import textwrap
 from collections.abc import Callable
 from datetime import date
+from typing import TypeVar
+
+import numpy as np
```

Comment:
Unless I am missing something, we can use our internal jax.numpy / numpy version. Code should be the same.

Reply:
Doesn't seem to work without changes. If I keep the code as is and use JAX, I get this error:

```txt
>           idx = backend.searchsorted(primary_key, foreign_key, sorter=sorter)
E           ValueError: Non-hashable static arguments are not supported. An error occurred while trying to hash an object of type <class 'jaxlib.xla_extension.ArrayImpl'>, [0 1 2]. The error was:
E           TypeError: unhashable type: 'jaxlib.xla_extension.ArrayImpl'
```

Reply:
Okay, we should keep that in mind then. We should do some benchmarks, but now that Jax is available on Windows, too, we may want to switch completely.